tests/pages/base_page.py

```
@@ -232,176 +242,142 @@ def _xpath_literal(s):
+                )
+
+                # Re-find the input after the option click because React/MUI can
+                # re-render it during selection.
+                def selection_finished(driver):
+                    try:
+                        current_input = driver.find_element(*input_locator)
+                        value = (current_input.get_attribute("value") or "").strip()
+                        expanded = current_input.get_attribute("aria-expanded")
+
+                        return (
+                            value == selected_text
```

Ruff (0.15.20)

[warning] Function definition does not bind loop variable `selected_text`

(B023)